### DISCUSSION TITLE
Herrington Darkholme - ast-grep

### DISCUSSION THREAD

**Discussion by just-be-dev** [devtools-fm/devtools.fm#270] [Announcements]
- https://ast-grep.github.io/
- https://ast-grep.github.io/playground.html
- https://twitter.com/hd_nvim

## Introduction

Andrew: Hello and welcome to DevTools.fm, a podcast about developer tools and the people who make them. I'm Andrew and this is my co-host Justin.

Justin: Hi everyone! In today's episode we're joined by Herrington Darkholme the creator of ast-grep. I've been using ast-grep a lot lately so I'm really excited to have you on Herrington. Before we dig in, would you like to tell our audience a little more about yourself?

### Background

- [x] You're working on a rust based vue compiler. Could you tell us a bit more about that?

### ast-grep

- [x] How would you describe ast-grep to someone who hasn't used a tool like this before?
- [x] What kinds of problems do you recommend people use it for?  What was the motivation behind building it?
- [x] What type of code can it handle? In the docs I noticed it mostly single node stuff, can I write rules for parent child relationship too?
- [x] Rules are written primarily in yaml. Why did you decide on that over a code based approach?
- [x] You've built a testing mechanism for rules. Could you explain a little more about how that works? (Also, why is in a separate file?)
- [x] There is an SDK for JS and Python. What are the limitations of using those SDKs vs just writing rules in yaml?
- [x] You're working on a VSCode extension. Could you tell us more about what that does and your plans for it?

### Comparisons

- [x] How does it compare to similar tools like comby or semgrep?
- [x] How does it compare with codemod tools like jscodeshift?

### Future

- [x] What do you have planned for the tool?
- [x] Do you think rust is the future of web based tooling?

**Comment by HerringtonDarkholme** [comment]
Glad to be here! This is a tooltip list of talk from me. 
https://workflowy.com/s/devtools-podcast/pANEflM0AYTBye0B

**Comment by hipstersmoothie** [comment]
https://github.com/DaniGuardiola/classy-ink
https://github.com/electric-sql/pglite

**Comment by just-be-dev** [comment]
https://github.com/lsd-rs/lsd
https://github.com/sindresorhus/ky
